cmd/notary-server/config.go

```diff
@@ -21,9 +23,21 @@ import (
 	_ "github.com/go-sql-driver/mysql"
 	"github.com/spf13/viper"
 	"golang.org/x/net/context"
-	"gopkg.in/dancannon/gorethink.v2"
+	gorethink "gopkg.in/dancannon/gorethink.v2"
 )
 
+// gets the required gun prefixes accepted by this server
+func getRequiredGunPrefixes(configuration *viper.Viper) ([]string, error) {
+	prefixes := configuration.GetStringSlice("repositories.gun_prefixes")
+	for _, prefix := range prefixes {
+		p := path.Clean(strings.TrimSpace(prefix))
+		if p+"/" != prefix || strings.HasPrefix(p, "/") || strings.HasPrefix(p, "..") {
```

Comment:
There's some pretty varied behavior for this currently in master, not sure if we want to separate fixing this into another PR.  Tested with notary from master and compose setup:

leading `/`:
```
$ bin/notary -c cmd/notary/config.json init /test/
Root key found, using: 2930402da13b3062f9f59cb3bc6f93d04ae0caaf5778318cf2de8a2eb123367b
Enter passphrase for root key with ID 2930402:
Enter passphrase for new targets key with ID 8ec423d (/test):
Repeat passphrase for new targets key with ID 8ec423d (/test):

* fatal: unable to reach trust server at this time: 301.
```

one level `..` escape:
```
$ bin/notary -c cmd/notary/config.json init ../test
Root key found, using: 2930402da13b3062f9f59cb3bc6f93d04ae0caaf5778318cf2de8a2eb123367b
Enter passphrase for root key with ID 2930402:
Enter passphrase for new targets key with ID 8fa72b4 (../test):
Repeat passphrase for new targets key with ID 8fa72b4 (../test):

* fatal: timestamp key trust data unavailable.  Has a notary repository been initialized?
```

multiple `..` escape
```
$ bin/notary -c cmd/notary/config.json init ../../test
Root key found, using: 2930402da13b3062f9f59cb3bc6f93d04ae0caaf5778318cf2de8a2eb123367b
Enter passphrase for root key with ID 2930402:
Enter passphrase for new targets key with ID c36d72b (../../test):
Repeat passphrase for new targets key with ID c36d72b (../../test):

* fatal: failed to add key to filestore: path outside file store
```